packages/server/src/queries/complex/assets/bridge.ts

```diff
@@ -0,0 +1,30 @@
+import { Asset as AssetListAsset, AssetList } from "@osmosis-labs/types";
+
+import { Asset } from ".";
+
+/** A bridgeable asset. */
+export type BridgeAsset = {
+  transferMethods: AssetListAsset["transferMethods"];
+  counterparty: AssetListAsset["counterparty"];
+};
+
+/** Appends bridge info to a given asset. If asset is not found in asset list, empty bridge info will be returned.
+ *  @throws if a given asset is not found in asset list.
+ */
+export function getBridgeAsset<TAsset extends Asset>(
+  assetLists: AssetList[],
+  asset: TAsset
+): TAsset & BridgeAsset {
+  const assetListAsset = assetLists
+    .flatMap(({ assets }) => assets)
+    .find((a) => a.coinMinimalDenom === asset.coinMinimalDenom);
```

Comment:
Nit:

```suggestion
  import { getAssetFromAssetList } from "@osmosis-labs/utils"

  const assetListAsset = getAssetFromAssetList({
     coinMinimalDenom: asset.coinMinimalDenom
  })
```

Reply:
See other comment

Reply:
Since we're not tying this to the `getAssetFromAssetList` method, adding the `getAssetFromAssetlist` here shouldn’t hinder our ability to switch to another data source later on. We’re including it mainly to take advantage of the `find` we already know works well. What do you think?

Reply:
True, but I like to avoid using utils such as those as it adds an unnecessary dependency that could be avoided. Also, as a new dev, I'm unaware of that `getAssetFromAssetList` does under the hood. Does it modify asset list data? I'm not sure, unless I jump into the implementation. With just a flatmap and a find, I know I'm just reading data from the asset list. It's more transparent and involves less indirection. I think we've discussed this before but that's my rationale.